source/simulator/stokes_matrix_free.cc

```diff
@@ -1515,14 +1515,28 @@ namespace aspect
           else
             {
               smoother_data_Schur[0].smoothing_range = 1e-3;
-              smoother_data_Schur[0].degree = numbers::invalid_unsigned_int;
-              smoother_data_Schur[0].eig_cg_n_iterations = 100; /*mg_matrices_M[0].m();*/
+              smoother_data_Schur[0].degree = 8;
+              smoother_data_Schur[0].eig_cg_n_iterations = 100;
             }
           smoother_data_Schur[level].preconditioner = mg_matrices_Schur_complement[level].get_matrix_diagonal_inverse();
         }
       mg_smoother_Schur.initialize(mg_matrices_Schur_complement, smoother_data_Schur);
     }
 
+#if DEAL_II_VERSION_GTE(9,2,0)
+    // Estimate the eigenvalues for the Chebyshev smoothers.
```

Comment:
Please extend the comment to say why you do this (is it faster? is it for error checking?). Even after our conversation I do not understand, but maybe that is just me :-)

Reply:
Eventually we would like to move the setup of the GMG preconditioner into an assembly timing block (probably "Build Stokes preconditioner") instead of begin contained within the "Solve Stokes system" timing block, and so this new feature in deal.II (being able to estimate the Chebyshev eigenvalues outside the smoother application) gets us closer to this. But there is no time save here.